astropy/utils/data.py

```diff
@@ -256,22 +290,48 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         # use configfile default
         remote_timeout = conf.remote_timeout
 
+    # Have `use_fsspec` default to ``True`` if the user passed an Amazon S3
+    # or Google Cloud Storage URI.
+    if use_fsspec is None and _requires_fsspec(name_or_obj):
+        use_fsspec = True
+
+    if use_fsspec:
+        if not isinstance(name_or_obj, str):
+            raise TypeError("`name_or_obj` must be a string when `use_fsspec=True`")
+        # For s3:// paths, it is useful to have fsspec default to `anon=True`
+        # because Hubble/JWST's data archive is available via a public S3 buckets.
+        # Accessing a public bucket without credentials raises a
+        # `NoCredentialsError` unless `anon=True` is passed explicitely.
+        if fsspec_kwargs is None:
+            if name_or_obj.startswith("s3://"):
+                fsspec_kwargs = {'anon': True}
+            else:
+                fsspec_kwargs = {}
+
     # name_or_obj could be an os.PathLike object
     if isinstance(name_or_obj, os.PathLike):
         name_or_obj = os.fspath(name_or_obj)
 
     # Get a file object to the content
     if isinstance(name_or_obj, str):
-        is_url = _is_url(name_or_obj)
-        if is_url:
-            name_or_obj = download_file(
-                name_or_obj, cache=cache, show_progress=show_progress,
-                timeout=remote_timeout, sources=sources,
-                http_headers=http_headers)
-        fileobj = io.FileIO(name_or_obj, 'r')
-        if is_url and not cache:
-            delete_fds.append(fileobj)
-        close_fds.append(fileobj)
+        # Use fsspec to open certain cloud-hosted files (e.g., AWS S3, Google Cloud Storage)
+        if use_fsspec or _requires_fsspec(name_or_obj):
```

Comment:
I think here we should trust the user, and not add `_requires_fsspec` again: here `use_fsspec=False` can only happen if they set it explicitly, so then let the pieces fall where they may.

Reply:
Agreed!

Reply:
Fixed via 7e7d668852e800d5fce640c020309ccbc7ea4607.  Thanks for the excellent feedback!